conversions/prefix_conversions.py

```diff
@@ -0,0 +1,111 @@
+"""
+Convert SI and Binary prefixes
+"""
+
+from enum import Enum
+from typing import Union
+
+
+class SI_Unit(Enum):
+
+    yotta = 24
+    zetta = 21
+    exa = 18
+    peta = 15
+    tera = 12
+    giga = 9
+    mega = 6
+    kilo = 3
+    hecto = 2
+    deca = 1
+    deci = -1
+    centi = -2
+    milli = -3
+    micro = -6
+    nano = -9
+    pico = -12
+    femto = -15
+    atto = -18
+    zepto = -21
+    yocto = -24
+
+
+class Binary_Unit(Enum):
+
+    yotta = 8
+    zetta = 7
+    exa = 6
+    peta = 5
+    tera = 4
+    giga = 3
+    mega = 2
+    kilo = 1
+
+
+def convert_si_prefix(
+    known_amount: float,
+    known_prefix: Union[str, SI_Unit],
+    unknown_prefix: Union[str, SI_Unit],
+) -> float:
+    """
+    Wikipedia reference: https://en.wikipedia.org/wiki/Binary_prefix
+    >>> convert_si_prefix(1, SI_Unit.giga, SI_Unit.mega)
+    1000
+    >>> convert_si_prefix(1, SI_Unit.mega, SI_Unit.giga)
+    0.001
+    >>> convert_si_prefix(1, SI_Unit.kilo, SI_Unit.kilo)
+    1
+    >>> convert_si_prefix(1, 'giga', 'mega')
+    1000
+    """
+
+    if type(known_prefix) is str:
+
+        known_prefix: SI_Unit = SI_Unit[known_prefix]
```

Comment:
```suggestion
        known_prefix: SI_Unit = SI_Unit[known_prefix.lower()]
```
and please add a test that uses mixed case or upper case.